api/v1/sotreus/sotreus.go

```
@@ -0,0 +1,252 @@
+package sotreus
+
+import (
+	"bytes"
+	"encoding/json"
+	"io"
+	"net/http"
+
+	"github.com/NetSepio/gateway/api/middleware/auth/paseto"
+	"github.com/NetSepio/gateway/config/dbconfig"
+	"github.com/NetSepio/gateway/config/envconfig"
+	"github.com/NetSepio/gateway/models"
+	"github.com/NetSepio/gateway/util/pkg/logwrapper"
+	"github.com/TheLazarusNetwork/go-helpers/httpo"
+	"github.com/gin-gonic/gin"
+)
+
+func ApplyRoutes(r *gin.RouterGroup) {
+	g := r.Group("/vpn")
+	{
+		g.Use(paseto.PASETO)
+		g.POST("", Deploy)
+		g.POST("/stop", Stop)
+		g.DELETE("", Delete)
+		g.POST("/start", Start)
+		g.GET("/all", AllDeployments)
+		g.GET("/", MyDeployments)
+	}
+}
+
+func Deploy(c *gin.Context) {
+	db := dbconfig.GetDb()
+	walletAddress := c.GetString(paseto.CTX_WALLET_ADDRES)
+	var req DeployRequest
+	err := c.BindJSON(&req)
+	if err != nil {
+		logwrapper.Errorf("failed to bind JSON: %s", err)
+		httpo.NewErrorResponse(http.StatusInternalServerError, "failed to create VPN").SendD(c)
+		return
+	}
+	var DeployerRequest DeployerCreateRequest
+	DeployerRequest.SotreusID = req.Name
+	ReqBodyBytes, err := json.Marshal(DeployerRequest)
+	if err != nil {
+		logwrapper.Errorf("failed to encode request: %s", err)
+		httpo.NewErrorResponse(http.StatusInternalServerError, "failed to create VPN").SendD(c)
+		return
+	}
+	contractReq, err := http.NewRequest(http.MethodPost, envconfig.EnvVars.VPN_DEPLOYER_API+"/sotreus", bytes.NewReader(ReqBodyBytes))
+	if err != nil {
+		logwrapper.Errorf("failed to send request: %s", err)
+		httpo.NewErrorResponse(http.StatusInternalServerError, "failed to create VPN").SendD(c)
+		return
+	}
+	client := &http.Client{}
+	resp, err := client.Do(contractReq)
+	if err != nil {
+		logwrapper.Errorf("failed to send request: %s", err)
+		httpo.NewErrorResponse(http.StatusInternalServerError, "failed to create VPN").SendD(c)
+		return
+	}
+	defer resp.Body.Close()
+
+	body, _ := io.ReadAll(resp.Body)
```

don't ignore error here, handle it